src/Npgsql/TypeHandlers/NumericHandlers/NumericHandler.cs

```diff
@@ -48,67 +48,61 @@ public override decimal Read(NpgsqlReadBuffer buf, int len, FieldDescription? fi
             var sign = buf.ReadUInt16();
 
             if (sign == SignNan)
-                ThrowSafeReadException(new InvalidCastException("Numeric NaN not supported by System.Decimal"), len -6);
+            {
+                if (len - 6 > 0)
+                    buf.Skip(len - 6);
```

Comment:
Do we still need to skip here? Unless I'm mistaken this becomes unnecessary because it's handled in reader.

Reply:
Skip removed from all handlers in case of exception.